client/src/GameState.ts

```diff
@@ -379,6 +383,19 @@ export class GameState {
         // look-ahead for the next tetromino state after movement
         const lookahead = movement(this.currentTetromino);
 
+        if (
+            lookahead.tiles.some(([row, _]) => {
+                return row >= BOARD_SIZE;
+            }) &&
+            this.playerId &&
```

Comment:
It seems like `this.playerId` will ignore player 0, should make it `this.playerId != null`